sklearn/linear_model/tests/test_sag.py

```diff
@@ -820,83 +818,6 @@ def test_binary_classifier_class_weight(csr_container):
     assert_almost_equal(clf2.intercept_, spintercept2, decimal=1)
 
 
-@pytest.mark.filterwarnings("ignore:The max_iter was reached")
-@pytest.mark.parametrize("csr_container", CSR_CONTAINERS)
-def test_multiclass_classifier_class_weight(csr_container):
```

Comment:
I assume that the decision to directly remove this test now instead of waiting for 1.7 with an ignore warning is that it's mostly redundant with `test_logistic_regression_class_weights`?

The only difference I see is that `test_logistic_regression_class_weights` does not test on sparse input data and maybe this can indeed be relevant because the sag/saga solvers have special branches for sparse data. But I agree that this way of testing is quite weak because the complex part happens in the `sag_sparse` function which is used by `LogisticRegression` internally so it's a bit of a tautological test.

Maybe we can expand the parametrization of `test_logistic_regression_class_weights` to also consider sparse data instead.